### PR TITLE
Manifest info name should be case insensitive

### PR DESCRIPTION
**Pull request by pragnya17**
Existing code has an overriden `Equals()` method that returns true even if the manifestInfo has differing cases. This PR updates the manifestInfoValidator error message to indicate that and also adds unit tests to validate the behavior.

### CODE REVIEW

File: test/Microsoft.Sbom.Api.Tests/Config/Validators/ManifestInfoValidatorTests.cs [hunk 1]
```diff
@@ -19,7 +19,6 @@ public class ManifestInfoValidatorTests
     [DataRow("randomName", "2.2")]
     [DataRow("randomName", "3.0")]
     [DataRow("SPDX", "randomVersion")]
-    [DataRow("spdx", "randomVersion")]
```
Comment (jlperkins) [inline test/Microsoft.Sbom.Api.Tests/Config/Validators/ManifestInfoValidatorTests.cs:22]: Why remove this? Just curious

Comment (pragnya17) [reply test/Microsoft.Sbom.Api.Tests/Config/Validators/ManifestInfoValidatorTests.cs:22]: Lowercase spdx should not throw, it should be case-insensitive. So I've removed it from this unit test and added the case to the unit test below instead.

Comment (jlperkins) [reply test/Microsoft.Sbom.Api.Tests/Config/Validators/ManifestInfoValidatorTests.cs:22]: Ah - I think this test was looking at spdx + 'randomversion', which I believe should still throw. Like the test case above, with SPDX + 'randomversion', it should throw because of the version, not the name. 

Just for completeness, I'd keep it, but I guess I don't feel that strongly. But we should be clear on what this unit test is trying to do

**Comment by jlperkins** [comment]
Do you mind linking to the Equals() call you reference in the description? Just for ease of review :)

File: test/Microsoft.Sbom.Api.Tests/Config/Validators/ManifestInfoValidatorTests.cs (hunk 1 above)
Comment (DaveTryon) [reply test/Microsoft.Sbom.Api.Tests/Config/Validators/ManifestInfoValidatorTests.cs:22]: Positive tests should cover all happy paths--negative tests can often have just 1 case for each path and don't need to be exhaustive. Put another way, when the next SPDX format comes along, we would need to update the positive tests but not the negative tests.

It can also be a good idea to include a comment that explains the error should the test fail. This comment gets added to the `DataRow`, then used as the `message` parameter of the `Assert` method in question. If you were to combine these ideas for this test, you'd end up with the something like this:

```
    [DataRow("randomName", "2.2", "Unsupported Name should throw")]
    [DataRow("SPDX", "randomVersion", "Unsupported Version should throw")]
    [TestMethod]
    public void InvalidManifestInfoThrows(string name, string version, string message)
    {
        var invalidManifestInfo = new ManifestInfo
        {
            Name = name,
            Version = version
        };

        IList<ManifestInfo> listOfManifestInfos = new List<ManifestInfo> { invalidManifestInfo };

        var validator = new ManifestInfoValidator(mockAssemblyConfig.Object);
        Assert.ThrowsException<ValidationArgException>(() => validator.ValidateInternal("property", listOfManifestInfos, null), message);
    }
```

Comment (jlperkins) [reply test/Microsoft.Sbom.Api.Tests/Config/Validators/ManifestInfoValidatorTests.cs:22]: Makes sense, TIL negative tests don't need to be exhaustive. Thanks for the thorough answer!

**Comment by pragnya17** [comment]
> Do you mind linking to the Equals() call you reference in the description? Just for ease of review :) Edit: nvm, [found it](https://github.com/microsoft/sbom-tool/blob/26bb1b00ca4b3e08a86336b65483a96c92d10487/src/Microsoft.Sbom.Contracts/Contracts/SbomSpecification.cs#L90)

Yup that's the one I was referring to
